pytensor/sparse/rewriting.py

```diff
@@ -179,7 +179,7 @@ def c_code(self, node, name, inputs, outputs, sub):
                   }}
                  }}
                 }}
-             """.format(**dict(locals(), **sub))
+             """
```

Comment:
I think the contents of `sub` are not accounted for

Below you did
```python
fail = sub["fail"]
rval = f"""
```

Reply:
I think `sub` was superfluous here: all the patterns in this f-string are already accounted by local variables